hub/core/chunk_engine/chunker.py

```diff
@@ -0,0 +1,100 @@
+import numpy as np
+from typing import Generator, Optional, List
+
+from hub.util.exceptions import ChunkSizeTooSmallError
+
+
+def generate_chunks(
+    content_bytes: bytes,
+    chunk_size: int,
+    bytes_left_in_last_chunk: int = 0,
+) -> Generator[bytes, None, None]:
+    """Generator function that chunks bytes.
+
+    Chunking is the process of taking the input `content_bytes` & breaking it up into a sequence of smaller bytes called "chunks".
+    The sizes of each chunk are <= `chunk_size`.
+
+    Example:
+        content_bytes = b"1234567890123"
+        chunk_size = 4
+        yields:
+            b"1234"
+            b"5678"
+            b"9012"
+            b"3"
+
+    Args:
+        content_bytes (bytes): Bytes object with the data to be chunked.
+        chunk_size (int): Each individual chunk will be assigned this many bytes maximum.
+        bytes_left_in_last_chunk (int): If chunks were created already, `bytes_left_in_last_chunk`
+            should be set to the `chunk_size - len(last_chunk)`. This is so the generator's
+            first output will be enough bytes to fill that chunk up to `chunk_size`.
+
+    Yields:
+        bytes: Chunk of the `content_bytes`. Will have length on the interval (1, `chunk_size`].
+
+    Raises:
+        ChunkSizeTooSmallError: If `chunk_size` <= 0
+        ValueError: If `bytes_left_in_last_chunk` < 0
+    """
+
+    # validate inputs
+    if chunk_size <= 0:
+        raise ChunkSizeTooSmallError()
```

Comment:
TIL that you can just do `raise E` instead of `raise E()`. A simple regex should be able to fix these in one go. We can do it in a separate PR if you want to cut scope

Reply:
would be nice to be a separate PR